libs/api/Verifier.cpp

```diff
@@ -242,36 +243,71 @@ ebpf_api_elf_disassemble_section(
     return 0;
 }
 
+struct guid_compare
+{
+    bool
+    operator()(const GUID& a, const GUID& b) const
+    {
+        return (memcmp(&a, &b, sizeof(GUID)) < 0);
+    }
+};
+
+thread_local std::map<GUID, ebpf_helper::ebpf_memory_ptr, guid_compare> g_program_information_cache;
+
+ebpf_result_t
+get_program_type_info(const ebpf_program_information_t** info)
+{
+    const GUID* program_type = (const GUID*)global_program_info.type.platform_specific_data;
```

Comment:
Suggestion: Use C++ style casts? This would be a reinterpret_cast. This makes the intent explicit (i.e. you are explicitly changing the type vs changing constness etc).

Reply:
done